packages/foam-vscode/src/core/utils/path.ts

```diff
@@ -0,0 +1,88 @@
+import { posix } from 'path';
+import { promises, constants } from 'fs';
+
+export function isIdentifier(uriPath: string): boolean {
+  return !(
+    uriPath.startsWith('/') ||
+    uriPath.startsWith('./') ||
+    uriPath.startsWith('../')
+  );
+}
+
+export function isAbsolute(uriPath: string): boolean {
+  return uriPath.startsWith('/');
+}
+
+export function relativeTo(baseUriPath: string, uriPath: string): string {
+  return posix.relative(posix.dirname(baseUriPath), uriPath);
+}
+
+export function getExtension(uriPath: string): string {
+  return posix.extname(uriPath);
+}
+
+export function removeExtension(uriPath: string): string {
+  let ext = getExtension(uriPath);
+  return uriPath.substring(0, uriPath.length - ext.length);
+}
+
+export function getBasename(uriPath: string): string {
+  return posix.basename(uriPath);
+}
+
+export function getName(uriPath: string): string {
+  return removeExtension(getBasename(uriPath));
+}
+
+export function getDir(uriPath: string): string {
+  return posix.dirname(uriPath);
+}
+
+export function joinPaths(baseUriPath: string, ...uriPaths: string[]): string {
+  return posix.join(baseUriPath, ...uriPaths);
+}
+
+export async function existsInFs(fsPath: string) {
+  try {
+    await promises.access(fsPath, constants.F_OK);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+export function isUNCShare(fsPath: string): boolean {
+  return fsPath.length >= 2 && fsPath[0] === '\\' && fsPath[1] === '\\';
+}
+
+export function parseUNCShare(uncPath: string): [string, string] {
+  const idx = uncPath.indexOf('\\', 2);
+  if (idx === -1) {
+    return [uncPath.substring(2), '\\'];
+  } else {
+    return [uncPath.substring(2, idx), uncPath.substring(idx) || '\\'];
+  }
+}
+
+export function toUriPath(path: string): string {
```

Comment:
A way to solve this by ignoring it is:
- Rename `toFsPath/toUriPath` to `toFsPath/fromFsPath` (moreover, this nicely mimics `toVsCodeUri/fromVsCodeUri`).
- Rename all `uriPath` arguments to just `path` on the understanding that `path` means platform-agnostic paths here (the ones in URIs, similar to POSIX).

There are some shortcomings though:
- The relationship between "path in uri.path" and "platform-agnostic path" is not conveyed by the new naming, although since now most path functionality is accessed through URI wrappers this is not as relevant as it was before.
- Some path functions can receive uriPaths as well as fsPaths (specifically, `toUriPath` is able to deal with both). In these cases `path` meant `uriPath|fsPath`. I can call this `anyPath` so as to reserve `path` to mean the former `uriPath`.

So we get:

- `toFsPath/fromFsPath`  to move to/from platform-specific paths.
- A `path` argument means platform-agnostic path.
- A `fsPath` argument means platform-specific path.
- A `uncPath` argument means UNC share.
- A `anyPath` argument means `path|fsPath`.

I prefer the old naming convention for arguments, but all in all I believe this change would be for the best.

Reply:
I've addressed this in a slightly different way since the `fromFsPath` name doesn't cut the mustard after all: the input is indeed an `anyPath`, in the sense of `path|fsPath` (for example, when parsing a URI both kind of paths may arrive). So the name is now simply `asPath`. The signature is `asPath(value: string): path`. This function prepares a path: normalizes drive letter case, replaces backslashes with slashes, etc. I'm still reluctant to call it anything posix-related because I think of it as an agnostic path, not a posix path. `asFoamPath` is another option, but `asPath` seems quite decent to me.